File: freeform-server/src/main/java/com/sunshine/freeform/MyLog.java
package com.sunshine.freeform;

import java.util.Date;

/**
 * @author sunshine
 * @date 2021/3/14
 */
public class MyLog {

    private static StringBuilder logBuilder = new StringBuilder();

    private static long MAX_SIZE = 100000;

    /**
     * 日志过多时应该适当清理
     */
    private static void needFree() {
        if (logBuilder.length() > MAX_SIZE) {
            logBuilder.delete(0, logBuilder.length());
            logBuilder.append("-----some log was delete because log is to big-----\n");
        }
    }

    public static void d(String log) {
        needFree();
        logBuilder.append(new Date(System.currentTimeMillis())).append("\ndebug:").append(log).append("\n");
    }

    public static void e(String log) {
        needFree();
        logBuilder.append(new Date(System.currentTimeMillis())).append("\nerror:").append(log).append("\n");
    }

    public static String getLog() {
        return logBuilder.toString();
    }
}

File: app/src/main/java/com/sunshine/freeform/utils/MyLog.kt
package com.sunshine.freeform.utils

import java.lang.StringBuilder

/**
 * @author sunshine
 * @date 2021/3/17
 */
object MyLog {

    private val log = StringBuilder()

    fun d(tag: String, whatFun: String, log: String) {
        this.log.append("$tag\$$whatFun:$log\n")
    }

    fun e(tag: String, whatFun: String, log: String) {
        this.log.append("$tag\$$whatFun:$log\n")
    }
}